ytdl_nfo/__init__.py

```diff
@@ -8,22 +9,25 @@ def main():
     parser.add_argument('input', metavar='JSON_FILE', type=str, help='Json file to convert or folder to convert in')
     parser.add_argument('-e', '--extractor', help='Specify specific extractor')
     parser.add_argument('-w', '--overwrite', action="store_true", help='Overwrite existing NFO files')
+    parser.add_argument('--regex', type=str, help='Specify regex search string to match files', default=r".json$")
     parser.add_argument('--config', help='Prints the path to the config directory', action='version', version=f'{get_config_path()}')
     args = parser.parse_args()
 
+    extractor_str = args.extractor if args.extractor != None else "file specific"
+
     if os.path.isfile(args.input):
-        print(f'Processing {args.input} with extractor {args.extractor}')
+        print(f'Processing {args.input} with {extractor_str} extractor')
         file = Ytdl_nfo(args.input, args.extractor)
         file.process()
+        file.write_nfo()
     else:
-        for root, dirs, files in os.walk(args.input):
-            for file_name in files:
-                file_path = os.path.join(root, file_name)
-                if file_name.endswith('.info.json'):
-                    if args.overwrite or not os.path.exists(file_path[:-10] + ".nfo"):
-                        print(f'Processing {file_path} with extractor {args.extractor}')
-                        file = Ytdl_nfo(file_path, args.extractor)
-                        file.process()
+        print(f"Processing Folder {args.input}")
+        for filename in os.listdir(args.input):
+            if re.search(args.regex, filename):
```

Comment:
Hey, doesn't this diff remove the recursive traversal functionality? I cannot use the tool on my video archive if it doesn't support recursive traversal.

Reply:
Looks like you also removed support for the overwrite flag, which is another requirement for me

Reply:
I think the diffs may have messed up because I had to do a 3 way merge to rebase dev back to master due to my earlier lack of git knowledge. If you check the latest release/dev branch the overwrite and recursive functionality should still be in there.

Reply:
Yeah looking through the diff log it seems that this commit reverted the walk and overwrite functionality and the next one re-implemented it. Bad code management on my part I'm sorry.

Reply:
No worries, I figured it was probably a merge related thing. As long as it finds its way back in for the release I've got no complaints.